FashionConnect_ui/src/components/Products/Products.jsx

```diff
@@ -1,29 +1,96 @@
 import "./Products.css"
-import { useState, useEffect, useContext } from "react";
+import React, { useState} from "react";
+import ProductCard from "../ProductCard/ProductCard";
 
-export default function Products(){
-    const [products, setProducts] = useState([]);
+export default function Products({products}){
+   
+    const [searched, setSearched] = useState("");
+    const regexp = new RegExp(searched, "i");
+    const [clickedCategory, setClickedCategory] = useState("");
 
-    useEffect(() => {
-        const fetchProducts = async () => {
-        const response = await fetch('http://localhost:3000/products');
-        const data = await response.json();
-        setProducts(data);
-        };
-        fetchProducts();
-    }, []);
-    return(
-        <div className="main">
-        <div className="products-container">
-        {products.map((product) => (
-          <div className="product" key={product.ProductID}>
-            <img src={product.Image_URL} alt={product.Name} className="product-image" />
-            <div className="product-name"><h3>{product.Name}</h3></div>
-            {/* <div className="product-description"><p className="description">{product.Description}</p></div> */}
-            <div className="product-price"><p className="price">{product.Price}</p></div>
-          </div>
-        ))}
-      </div>
-      </div>
+    const handleClick = (category) => {
+        console.log(category)
+      setClickedCategory(category);
+    };
+  
+    const handleSearched = (event) => {
+      const searched = event.target.value;
+      setSearched(searched);
+    };
+  
+    let currProds = clickedCategory ? products.filter((product) => product.category.Name === clickedCategory):products
+    let currSearch = currProds.filter((product) => regexp.test(product.Name));
+  
+    return (
+      <section className="products">
+         <div className="search-comp">
+            <form className= "search-text-input" method="get-search-input" onChange={(event) => handleSearched(event)}>
+                        <input className="search-text" type="text" required />
+                    <div className="button" id="s-cover">
+                        <button className="search" type="submit">SEARCH
+                        </button>
+                </div>
+            </form>
+        </div>
+        <div className="menu">
+            <li>
+            <button className="menu-btn" onClick={() => handleClick("")}>
+                All Categories
+            </button>
+            </li>
+            <li>
+            <button className="menu-btn" onClick={() => handleClick("Tops")}>
```

Comment:
Once you have the data available in your database, you should replace this strings with the categoryID. That way, you can turn this entire logic into function calls that returns the button component with the right text.

Reply:
You can also consider doing this now too with the strings and see how your code will shrink to a few lines. :)

Reply:
Ok cool!
I already have the data for caretgory Id. I will apply the changes